bin/github-review.sh

```diff
@@ -45,53 +41,130 @@ if ! [[ $1 =~ ^[0-9]+$ ]]; then
+    diffPatchSeparated=($(gh pr diff $pr | grep -E '^[+-] '))
+    # Adding three elements at a time together to compare from the output array
+    diffPatch=()
+    for ((i=0; i<${#diffPatchSeparated[@]}; i+=3)); do
+        diffPatch+=("${diffPatchSeparated[i]} ${diffPatchSeparated[i+1]} ${diffPatchSeparated[i+2]}")
+    done
+
+    # Initialize a counter
+    counter=1
+    # Initialize a flag to indicate if the previous pattern was correct
+    pattern_correct=true
+    # Read input line by line from the array
+    for line in "${diffPatch[@]}"; do
+        # Determine the line type based on the counter
+        case $((counter % 3)) in
+            1) # First line should start with '-'
+                if [[ $line != -* ]]; then
+                    pattern_correct=false
+                fi
+                # Store the first line to compare with the second line
+                first_line="$line"
+                ;;
+            2) # Second line should be the first line with a comma
+                expected_line="+${first_line:1},"
+                if [[ $line != $expected_line ]]; then
+                    pattern_correct=false
+                fi
+                ;;
+            0) # Third line should start with '+'
+                if [[ $line != +* ]]; then
+                    pattern_correct=false
+                fi
+                # Check the pattern for the set of three lines
+                if [ "$pattern_correct" = false ]; then
+                    echo "Unknown lines added or removed" 1>&2
+                    exit 1
+                fi
+                # Reset the pattern_correct flag for the next set of lines
+                pattern_correct=true
+                ;;
+        esac
+        # Increment the counter
+        ((counter++))
+    done
+fi
+
 echo "Verifying Deployment Asset"
-gh pr diff $pr --patch | git apply --include 'src/assets/**'
+gh pr diff $pr --patch | git apply --include 'src/assets/v'$version'/**'
 
 # Getting default addresses, address on the chain and checking code hash.
+deploymentTypes=($(jq -r --arg c "$chainid" '.networkAddresses[$c]' "$versionFiles"))
+if [[ $deploymentTypes == "[" ]]; then
+    deploymentTypes=($(jq -r --arg c "$chainid" '.networkAddresses[$c][]' "$versionFiles"))
+fi
 for file in "${versionFiles[@]}"; do
-    defaultAddress=$(jq -r '.defaultAddress' "$file")
-    networkAddress=$(jq -r --arg c "$chainid" '.networkAddresses[$c]' "$file")
-    if [[ $defaultAddress != $networkAddress ]]; then
-        echo "ERROR: "$file" default address is not the same as the one created for the chain id" 1>&2
-        exit 1
-    fi
-    defaultCodeHash=$(jq -r '.codeHash' "$file")
-    networkCodeHash=$(cast keccak $(cast code $networkAddress --rpc-url $rpc))
-    if [[ $defaultCodeHash != $networkCodeHash ]]; then
-        echo "ERROR: "$file" code hash is not the same as the one created for the chain id" 1>&2
-        exit 1
-    fi
+    for deploymentType in "${deploymentTypes[@]}"; do
```

Comment:
Doesn't this run for each version type for each file for each file? My understanding is that the `deploymentTypes` is built for **all** files (so it will have a duplicate for each file).

Reply:
Yes, it will run for each version if there is more than one version deployed for that particular chain (only limited to `v1.3.1`). Which I think should not be a problem, other than if the RPC have some limit of less than 18 requests per minute or so.

Reply:
This can be fixed by deduping `deploymentTypes`. My point is that, AFAIU from how JQ works with multiple files specified, for 1.3.1 it will be something like:

```
deploymentTypes=(canonical eip155 canonical eip155 canonical eip155 canonical eip155 canonical eip155 canonical eip155 canonical eip155 canonical eip155 canonical eip155)
```

Reply:
Oh... I guess I don't know bash arrays very well. Is `"$versionFiles"` implicitly doing `"${versionFiles[0]}"`?

Reply:
I believe yes: https://stackoverflow.com/a/47368011/7520013